test/controllers/chargeD/CheckYourAnswersControllerSpec.scala

```diff
@@ -21,40 +21,103 @@ import controllers.base.ControllerSpecBase
+
+    (isPSR, isChargeInAddition, wasAnotherPensionScheme) match {
+      case (false, _, _) => ua.setOrException(IsPublicServicePensionsRemedyPage(lifetimeAllowanceCharge, 0), isPSR)
+      case (true, false, _) =>
+        ua.setOrException(IsPublicServicePensionsRemedyPage(lifetimeAllowanceCharge, 0), isPSR)
+          .setOrException(IsChargeInAdditionReportedPage(lifetimeAllowanceCharge, 0), isChargeInAddition)
+      case (true, true, false) =>
+        ua.setOrException(IsPublicServicePensionsRemedyPage(lifetimeAllowanceCharge, 0), isPSR)
+          .setOrException(IsChargeInAdditionReportedPage(lifetimeAllowanceCharge, 0), isChargeInAddition)
+          .setOrException(WasAnotherPensionSchemePage(lifetimeAllowanceCharge, 0), wasAnotherPensionScheme)
+          .setOrException(TaxYearReportedAndPaidPage(lifetimeAllowanceCharge, 0, None), dynamicYearRange)
+          .setOrException(TaxQuarterReportedAndPaidPage(lifetimeAllowanceCharge, 0, None), taxQuarter)
+          .setOrException(ChargeAmountReportedPage(lifetimeAllowanceCharge, 0, None), chargeAmountReported)
+      case (true, true, true) =>
+        ua.setOrException(IsPublicServicePensionsRemedyPage(lifetimeAllowanceCharge, 0), isPSR)
+          .setOrException(IsChargeInAdditionReportedPage(lifetimeAllowanceCharge, 0), isChargeInAddition)
+          .setOrException(WasAnotherPensionSchemePage(lifetimeAllowanceCharge, 0), wasAnotherPensionScheme)
+          .setOrException(EnterPstrPage(lifetimeAllowanceCharge, 0, 0), pstrNumber)
+          .setOrException(TaxYearReportedAndPaidPage(lifetimeAllowanceCharge, 0, Some(0)), dynamicYearRange)
+          .setOrException(TaxQuarterReportedAndPaidPage(lifetimeAllowanceCharge, 0, Some(0)), taxQuarter)
+          .setOrException(ChargeAmountReportedPage(lifetimeAllowanceCharge, 0, Some(0)), chargeAmountReported)
+    }
+  }
+
+  private def rows(isPSR: Boolean, isChargeInAddition: Boolean, wasAnotherPensionScheme: Boolean) = Seq(
+    helper.isPsprForChargeD(0, Some(isPSR)),
     helper.chargeDMemberDetails(0, memberDetails),
     helper.chargeDDetails(0, chargeDDetails),
-    Seq(helper.total(chargeAmount1 + chargeAmount2))
+    Seq(helper.total(chargeAmount1 + chargeAmount2)),
+    helper.psprChargeDDetails(0, psprSummary(isPSR, isChargeInAddition, wasAnotherPensionScheme)).getOrElse(None),
+    helper.psprSchemesChargeDDetails(0, psprSummary(isPSR, isChargeInAddition, wasAnotherPensionScheme), wasAnotherPensionScheme)
   ).flatten
 
-  private val jsonToPassToTemplate: JsObject = Json.obj(
-    "list" -> rows
+  private def jsonToPassToTemplate(isPSR: Boolean, isChargeInAddition: Boolean, wasAnotherPensionScheme: Boolean): JsObject = Json.obj(
+    "list" -> rows(isPSR, isChargeInAddition, wasAnotherPensionScheme)
   )
 
-  "CheckYourAnswers Controller" must {
+  "CheckYourAnswers Controller for (false, false, false)" must {
```

Comment:
can we give a proper name for this test ?

Reply:
done